packages/mui-docs/src/CodeCopy/CodeCopy.tsx

```diff
@@ -2,8 +2,8 @@ import * as React from 'react';
 import { useRouter } from 'next/router';
 import clipboardCopy from 'clipboard-copy';
 
-const CodeBlockContext = React.createContext<React.MutableRefObject<HTMLDivElement | null>>({
-  current: null,
```

Comment:
when `ref` is defined with null, react treats ref as unmutable. hence changed null to undefined

Reply:
Is this documented somewhere?

Reply:
In this case, using `React.useRef<number | null>(null)` makes the ref mutable.

Reply:
okay my bad, i misunderstood. 

as of now, react-compiler throws error when modifying ref inside event handler. same issue was reported here https://github.com/facebook/react/issues/29106. looks like it is treated as bug (https://github.com/facebook/react/issues/29106#issuecomment-2131486648) and fixed here https://github.com/facebook/react/pull/29591. once fix gets published and we update `eslint-plugin-react-complier` i'm expecting this error to go away. hence reverted all changes done in this file